File: app/src/main/java/org/thoughtcrime/securesms/messagerequests/GroupInfo.java
package org.thoughtcrime.securesms.messagerequests;

import androidx.annotation.NonNull;

public final class GroupInfo {
  public static final GroupInfo ZERO = new GroupInfo(0, 0, "");

  private final int    fullMemberCount;
  private final int    pendingMemberCount;
  private final String description;

  public GroupInfo(int fullMemberCount, int pendingMemberCount, @NonNull String description) {
    this.fullMemberCount    = fullMemberCount;
    this.pendingMemberCount = pendingMemberCount;
    this.description        = description;
  }

  public int getFullMemberCount() {
    return fullMemberCount;
  }

  public int getPendingMemberCount() {
    return pendingMemberCount;
  }

  public @NonNull String getDescription() {
    return description;
  }
}

File: app/src/main/java/org/thoughtcrime/securesms/messagerequests/GroupInfo.kt
package org.thoughtcrime.securesms.messagerequests

import org.thoughtcrime.securesms.recipients.Recipient

/**
 * Group info needed to show message request state UX.
 */
class GroupInfo(
  val fullMemberCount: Int = 0,
  val pendingMemberCount: Int = 0,
  val description: String = "",
  val hasExistingContacts: Boolean = false,
  val membersPreview: List<Recipient> = emptyList()
) {
  companion object {
    @JvmField
    val ZERO = GroupInfo()
  }
}
